### ISSUE TITLE
Training Performance Apple Silicon MPS

### ISSUE THREAD

**Issue by mmoollllee** [ultralytics/ultralytics#22540]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello :)

I hope this question is not considered to be in the wrong place here. I'm posting it, because I think it could be valuable for others too.

Is there any performance comparison for training large object detection datasets using ultralytics on Apple Silicon?
I'm wondering if it could be worth the try to run a training on a local machine (M3 Pro with 36GB RAM) instead of a cloud GPU (8-Core CPU, 48 GB RAM, 1x L4 24 GB VRAM). What could be expected results regarding training time? Are there any differences in the results to be expected?
With my dataset, training for about 100 epochs took about 48h on the mentioned cloud GPU.

Maybe there's a rule of thumb, like 1/2 the performance or maybe even equal on apple silicon?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @mmoollllee, thanks for opening this discussion and for using Ultralytics 🚀! This is an automated response to help you get started quickly—an Ultralytics engineer will also follow up here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question (like performance comparisons), please share more details to help the team and community provide accurate guidance:
- Exact Ultralytics version and PyTorch version
- Hardware details (Apple Silicon chip model, memory), whether MPS is enabled, and any system constraints
- Dataset size (#images), image resolution, and augmentation settings
- Model and task (e.g., YOLO11s detection), epochs, batch size, img size, optimizer/hyp settings
- Training logs snippets (start banner, scaler/MPS messages, speed per iter/epoch)
- Any differences you observed between environments and how you measured them ⏱️

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Providing the above details will help us compare your Apple Silicon MPS training against your cloud L4 setup and share relevant expectations or tips. 💡

**Comment by glenn-jocher** [comment]
Thanks—Apple Silicon (MPS) training is supported and stable, but for large detection datasets an NVIDIA L4 will generally train notably faster; accuracy should be comparable across devices (differences are usually just seed variance). For a quick, data‑specific estimate, time one epoch on both machines with identical settings and compare the reported s/epoch:

```bash
# M3 Pro (MPS)
yolo detect train data=path/to/data.yaml model=yolo11m.pt epochs=1 imgsz=640 device=mps

# L4 (CUDA)
yolo detect train data=path/to/data.yaml model=yolo11m.pt epochs=1 imgsz=640 device=0
```

If you can share those one‑epoch times, we can help interpret; setup details for Macs are in the [Apple Silicon MPS training guide](https://docs.ultralytics.com/modes/train/#apple-silicon-mps-training).
